### PR TITLE
replaced bookrepository with jparepository, added Liquibase support a…

### PR DESCRIPTION
**Pull request by Nikname2303**
…nd implemented soft delete

### CODE REVIEW

File: src/main/java/com/example/bookshop/controller/BookController.java
Comment (fedaktaras) [inline src/main/java/com/example/bookshop/controller/BookController.java:1]: PUT: /api/books/{id} (Update a specific book) 
Update required

File: src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java
Comment (fedaktaras) [inline src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java:1]: Update required

File: src/main/resources/liquibase.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+url=jdbc:mysql://localhost:3306/book_shop
+username=root
+password=MySQL1234
```
Comment (ValentynSharshon) [inline src/main/resources/liquibase.properties]: As I know, it's bad practice to pass the login and password to connect to the database in the configuration files.

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -4,5 +4,5 @@ spring.datasource.username=root
 spring.datasource.password=MySQL1234
```
Comment (ValentynSharshon) [inline src/main/resources/application.properties]: As I know, it's bad practice to pass the login and password to connect to the database in the configuration files.

File: src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java
Comment (ValentynSharshon) [reply src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java:1]: I agree with Taras.

**Review by ValentynSharshon — CHANGES_REQUESTED** [review]
Good job, but the task is not fully completed.

File: src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -37,4 +38,38 @@ public BookDto findById(Long id) {
...
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public void updateById(Long id, CreateBookRequestDto requestDto) {
+        Optional<Book> optionalBook = bookRepository.findById(id);
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java]: use orElseThrow()

File: src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -37,4 +38,38 @@ public BookDto findById(Long id) {
...
+                book.setPrice(requestDto.getPrice());
+            }
+            if (requestDto.getIsbn() != null) {
+                book.setIsbn(requestDto.getIsbn());
+            }
+            if (requestDto.getTitle() != null) {
+                book.setTitle(requestDto.getTitle());
+            }
+            if (requestDto.getAuthor() != null) {
+                book.setAuthor(requestDto.getAuthor());
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java]: let's remove these checks. We will add validation in the next HW. For now it is ok to remove those

File: src/main/resources/application.properties (hunk 2 above)
Comment (yevheniisynytsia) [reply src/main/resources/application.properties]: It is not very good to push those to github. Bad habit

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        - createTable:
+            tableName: books
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
```
Comment (yevheniisynytsia) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]:
```suggestion
```

Comment (yevheniisynytsia) [reply src/main/resources/db/changelog/changes/01-create-books-table.yaml]: primaryKey is not nullable by default

File: src/main/resources/db/changelog/changes/02-add-column-isDeleted-to-books-table.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  - changeSet:
+      id: add-column-isDeleted-to-books-table
+      author: nikname2303
+      changes:
+        - addColumn:
+            tableName: books
+            columns:
+              - column:
+                  name: is_deleted
+                  type: BIT
```
Comment (yevheniisynytsia) [inline src/main/resources/db/changelog/changes/02-add-column-isDeleted-to-books-table.yaml:11]:
```suggestion
                  type: boolean
```

Comment (yevheniisynytsia) [reply src/main/resources/db/changelog/changes/02-add-column-isDeleted-to-books-table.yaml:11]: and add default value false

File: src/main/resources/liquibase.properties [hunk 7]
```diff
@@ -0,0 +1,4 @@
+url=jdbc:mysql://localhost:3306/book_shop
```
Comment (yevheniisynytsia) [inline src/main/resources/liquibase.properties]: liquibase should work without it by default

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/changes/02-add-column-isDeleted-to-books-table.yaml (hunk 6 above)
Comment (Nikname2303) [reply src/main/resources/db/changelog/changes/02-add-column-isDeleted-to-books-table.yaml:11]: Doesn`t work with boolean or BOOLEAN, make tinyint(1)

**Review by andrii-hoienko — APPROVED** [review]
Nice

**Pull request merged**